e2e/testcases/ignore_mutation_test.go

```diff
@@ -0,0 +1,261 @@
+	"kpt.dev/configsync/pkg/metadata"
+	"kpt.dev/configsync/pkg/reconcilermanager"
+	kstatus "sigs.k8s.io/cli-utils/pkg/kstatus/status"
+)
+
+func TestAddIgnoreMutationObject(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.DriftControl, ntopts.SyncWithGitSource(nomostest.DefaultRootSyncID, ntopts.Unstructured))
+	rootSyncGitRepo := nt.SyncSourceGitReadWriteRepository(nomostest.DefaultRootSyncID)
+
+	nt.T.Log("Adding a new namespace")
+	namespace := k8sobjects.NamespaceObject("bookstore", core.Annotation("season", "summer"))
+	nt.Must(rootSyncGitRepo.Add("acme/ns.yaml", namespace))
+	nt.Must(rootSyncGitRepo.CommitAndPush("add a namespace"))
+	nt.Must(nt.WatchForAllSyncs())
+
+	nt.T.Log("Add the ignore mutation to the namespace")
+	updatedNamespace := k8sobjects.NamespaceObject("bookstore", core.Annotation(metadata.LifecycleMutationAnnotation, metadata.IgnoreMutation))
+	nt.Must(rootSyncGitRepo.Add("acme/ns.yaml", updatedNamespace))
+	nt.Must(rootSyncGitRepo.CommitAndPush("update namespace"))
+	nt.Must(nt.Watcher.WatchObject(kinds.Namespace(), "bookstore", "",
+		testwatcher.WatchPredicates(
+			testpredicates.HasAnnotation("season", "summer"),
+			testpredicates.HasAnnotationKey(metadata.LifecycleMutationAnnotation))))
+}
+
+func TestDeclareIgnoreMutationForUnmanagedObject(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.DriftControl, ntopts.SyncWithGitSource(nomostest.DefaultRootSyncID, ntopts.Unstructured))
+	rootSyncGitRepo := nt.SyncSourceGitReadWriteRepository(nomostest.DefaultRootSyncID)
+
+	nt.T.Log("Add an unmanaged namespace using kubectl")
+	nsObj := k8sobjects.NamespaceObject("bookstore")
+	nt.Must(nt.KubeClient.Apply(nsObj))
+
+	if err := nt.Validate(nsObj.Name, "", &corev1.Namespace{}); err != nil {
+		nt.T.Error(err)
+	}
+
+	nt.T.Log("Declare the unmanaged namespace with the ignore mutation annotation and other spec changes")
+	namespace := k8sobjects.NamespaceObject(
+		nsObj.Name,
+		core.Annotation(metadata.LifecycleMutationAnnotation, metadata.IgnoreMutation),
+		core.Annotation("season", "summer"))
+	nt.Must(rootSyncGitRepo.Add("acme/ns.yaml", namespace))
+	nt.Must(rootSyncGitRepo.CommitAndPush("add a namespace"))
+	nt.Must(nt.Watcher.WatchObject(kinds.Namespace(), nsObj.Name, "",
+		testwatcher.WatchPredicates(
+			testpredicates.HasAnnotation(metadata.LifecycleMutationAnnotation, metadata.IgnoreMutation),
+			testpredicates.MissingAnnotation("season"))))
+}
+
+func TestDeclareExistingObjectWithIgnoreAnnotation(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.DriftControl, ntopts.SyncWithGitSource(nomostest.DefaultRootSyncID, ntopts.Unstructured))
+	rootSyncGitRepo := nt.SyncSourceGitReadWriteRepository(nomostest.DefaultRootSyncID)
+
+	nt.T.Log("Add an umanaged namespace with the ignore-mutation annotation using kubectl ")
+	nsObj := k8sobjects.NamespaceObject("bookstore",
+		core.Annotation(metadata.LifecycleMutationAnnotation, metadata.IgnoreMutation),
+	)
+
+	nt.Must(nt.KubeClient.Apply(nsObj))
```

Comment:
I addressed [this comment](https://github.com/GoogleContainerTools/kpt-config-sync/pull/1486/files/3bb9a0ad08e60bb96d435660685ce6d33c980830#r1838504954) here @nan-yu. However, when I run the test, it now fails with the ignore-mutation not being removed from the namespace on the cluster. It seems like it's because the ignore mutation annotation is adopted by the nomostest field manager on `nt.KubeClient.Apply`. How should the updated applier code handle this case?